tasks/delete_old_reconcile_jobs/delete_old_reconcile_jobs.py

```diff
@@ -0,0 +1,248 @@
+OS_ENVIRON_INTERNAL_RECONCILIATION_EXPIRATION_DAYS = (
+    "INTERNAL_RECONCILIATION_EXPIRATION_DAYS"
+)
+
+LOGGER = CumulusLogger(name="ORCA")
+
+
+def task(
+    internal_reconciliation_expiration_days: int,
+    db_connect_info: Dict,
+) -> None:
+    """
+    Gets all jobs older than internal_reconciliation_expiration_days days, then deletes their records in Postgres.
+
+    Args:
+        internal_reconciliation_expiration_days: Only reports updated before this many days ago will be deleted.
+        db_connect_info: See shared_db.py's get_configuration for further details.
+    """
+
+    user_engine = shared_db.get_user_connection(db_connect_info)
+    old_job_ids = get_jobs_older_than_x_days(
+        internal_reconciliation_expiration_days, user_engine
+    )
+
+    if old_job_ids is None:
+        LOGGER.info("No jobs found.")
+        return
+
+    delete_jobs(
+        old_job_ids,
+        user_engine,
+    )
+
+
+@shared_db.retry_operational_error()
+def get_jobs_older_than_x_days(
+    internal_reconciliation_expiration_days: int, engine: Engine
+) -> List[int]:
+    """
+    Gets all jobs older than internal_reconciliation_expiration_days days.
+    If none are found, returns null.
+
+    Args:
+        internal_reconciliation_expiration_days: Only reports updated before this many days ago will be retrieved.
+        engine: The sqlalchemy engine to use for contacting the database.
+
+    Returns: A list of ids for the jobs, or null if none were found.
+
+    """
+    try:
+        LOGGER.debug(
+            f"Getting jobs older than {internal_reconciliation_expiration_days} days."
+        )
+        with engine.begin() as connection:
+            sql_results = connection.execute(
+                # populate reconcile_phantom_report with files in orca.files but NOT reconcile_s3_object
+                get_jobs_sql(),
+                [
+                    {
+                        "internal_reconciliation_expiration_days": internal_reconciliation_expiration_days
+                    }
+                ],
+            )
+            return sql_results.fetchone()[
+                0
+            ]  # fetchone returns row. [0] returns the list of ids.
+    except Exception as sql_ex:
+        LOGGER.error(f"Error while getting jobs: {sql_ex}")
+        raise
+
+
+@shared_db.retry_operational_error()
+def delete_jobs(job_ids: List[int], engine: Engine) -> None:
+    """
+    Deletes all records for the given job ids from the database.
+
+    Args:
+        job_ids: The ids of the jobs containing s3 inventory info.
+        engine: The sqlalchemy engine to use for contacting the database.
+    """
+    try:
+        LOGGER.debug(f"Deleting data for job ids {job_ids}.")
+        with engine.begin() as connection:
+            connection.execute(
+                # populate reconcile_phantom_report with files in orca.files but NOT reconcile_s3_object
+                delete_jobs_sql(),
```

Comment:
See comment on DELETE SQL and possibly breaking them up. It would look like the following under the with block. Note below is not production code just an example.

```python
filter = {"internal_reconciliation_expiration_days": internal_reconciliation_expiration_days}

with engine.begin() as connection:
    # Add logging and or timing
    log.debug("Preparing to delete records from mismatch report.")
    # Start the timer
    start = time.perf_counter()
    # Delete the records and return the count
    r1 = connection.execute(delete_mismatch_report(), filter).fetchone()
    # Stop the timer
    stop = time.perf_counter()
    log.info(f"DELETED {r1} records from mismatch report in {stop - start} seconds.")
    connection.execute(delete_orphan_report(), filter)
    connection.execute(delete_phantom_report(), filter)
    connection.execute(delete_s3_report(), filter)
    connection.execute(delete_job(), filter)
```

Information on timing items in code can be found here https://realpython.com/python-timer/#python-timers

Information on returning record count can be found here https://docs.sqlalchemy.org/en/14/core/connections.html?highlight=rowcount#sqlalchemy.engine.CursorResult.rowcount , here https://stackoverflow.com/questions/2251567/how-to-get-the-number-of-deleted-rows-in-postgresql and here https://www.postgresql.org/docs/current/sql-delete.html